### PR TITLE
Add some initial disk performance alerts

### LINKED ISSUE #1302: [1] Add alerting for slow storage and timeouts
**Is your feature request related to a problem? Please describe.**
We want to be able to support scenarios where we use local disks for applications to allow for better performance, therefore we should also monitor the performance of this setup so we can take actions if needed.

**Describe the solution you'd like**
I'd like to see alerts for increased disk latency and for disk timeouts so that we can monitor this better.

This should be configurable so that whenever we have local disks provisioned for a node we should be able to target it specifically, since different type of storage will have different type of performance classes.

It could probably be generalised so we can monitor remote volumes to identify problems with those as well.

Node exporter already provides some metrics for this which may be enough.

**Definition of done:**
Configurable alerts for storage performance.

### PR DESCRIPTION
**Pull request by Zash**
> [!warning]
> **This is public repository, ensure not to disclose:**
> - [ ] personal data beyond what is necessary for interacting with this pull request
> - [ ] business confidential information, such as customer names

### What kind of PR is this?

**Required**: Mark one of the following that is applicable:

- [x] kind/feature       
- [ ] kind/improvement   
- [ ] kind/deprecation   
- [ ] kind/documentation 
- [ ] kind/clean-up      
- [ ] kind/bug           
- [ ] kind/other         

*Optional*: Mark one or more of the following that are applicable:

> [!important]
> Breaking changes should be marked `kind/admin-change` or `kind/dev-change` depending on type
> Critical security fixes should be marked with `kind/security`

- [ ] kind/admin-change  
- [ ] kind/dev-change    
- [ ] kind/security      
- [ ] [kind/adr]()       

### What does this PR do / why do we need this PR?

...

- Fixes #1302

#### Checklist

- [x] Proper commit message prefix on all commits
  
- Change checks:
  - [x] The change is transparent
  - [ ] The change is disruptive
  - [x] The change requires no migration steps
  - [ ] The change requires migration steps
- Metrics checks:
  - [x] The metrics are still exposed and present in Grafana after the change
  - [x] The metrics names didn't change (Grafana dashboards and Prometheus alerts are not affected)
  - [ ] The metrics names did change (Grafana dashboards and Prometheus alerts were fixed)
- Logs checks:
  - [ ] The logs do not show any errors after the change
- Network Policy checks:
  - [ ] Any changed pod is covered by Network Policies
  - [ ] The change does not cause any dropped packages in the `NetworkPolicy Dashboard`
- Pod Security Policy checks:
  - [ ] Any changed pod is covered by Pod Security Admission
  - [ ] Any changed pod is covered by Gatekeeper Pod Security Policies
  - [ ] The change does not cause any pods to be blocked by Pod Security Admission or Policies
- Falco checks:
  - [ ] The change does not cause any alerts to be generated by Falco
- Audit checks:
  - [ ] The change does not cause any unnecessary Kubernetes audit events
  - [ ] The change requires changes to Kubernetes audit policy

### CODE REVIEW

File: helmfile/charts/prometheus-alerts/files/disk-perf.yaml
Comment (Xartos) [inline helmfile/charts/prometheus-alerts/files/disk-perf.yaml:9]: Unnecessary comments
```suggestion
```

File: helmfile/charts/prometheus-alerts/files/disk-perf.yaml
Comment (Xartos) [inline helmfile/charts/prometheus-alerts/files/disk-perf.yaml:14]: Is there also an equivalent metric for write?

Comment (Zash) [reply helmfile/charts/prometheus-alerts/files/disk-perf.yaml:12]: Yes, added https://github.com/elastisys/compliantkubernetes-apps/pull/1799/commits/a66b05d92dc8c0b44718c6739a406febd32c332a

File: helmfile/charts/prometheus-alerts/files/disk-perf.yaml
Comment (Zash) [reply helmfile/charts/prometheus-alerts/files/disk-perf.yaml]: https://github.com/elastisys/compliantkubernetes-apps/pull/1799/commits/425b60c2a20fb31ef4059b59d264b8a914eb6a94

**Review by Xartos — APPROVED** [review]

File: helmfile/charts/prometheus-alerts/files/disk-perf.yaml
Comment (crssnd) [inline helmfile/charts/prometheus-alerts/files/disk-perf.yaml:6]:
```suggestion
          description: Disk {{ $labels.device }} Wait Time on {{ $labels.instance }} is {{ $value }}, check the workload
```
unless you want to make the rule threshold configurable from the overwrite configs, I would not keep the `adjust this alert`

File: helmfile/charts/prometheus-alerts/files/disk-perf.yaml
Comment (crssnd) [inline helmfile/charts/prometheus-alerts/files/disk-perf.yaml:16]: this comments if for all the alerts.
How did you decide on the thresholds and the duration (for: 30s)? 
Looking at my dev cluster the `DiskQueueSizeHigh` would generate a lot of alerts that will open and close fast.

File: helmfile/charts/prometheus-alerts/values.yaml
Comment (crssnd) [inline helmfile/charts/prometheus-alerts/values.yaml:121]: if we want this enabled by default you should disable it for wc, as only the admins should see and act on these
adding `defaultRules.rules.diskPerf: true` [here](https://github.com/elastisys/compliantkubernetes-apps/blob/main/helmfile/values/prometheus-user-alerts-wc.yaml.gotmpl) should be enough

File: helmfile/charts/prometheus-alerts/files/disk-perf.yaml
Comment (crssnd) [reply helmfile/charts/prometheus-alerts/files/disk-perf.yaml:16]: ok, can you add the option to completely disable them from the overwrite configs?
it would be good to have this option in case they are too noisy

File: helmfile/charts/prometheus-alerts/files/disk-perf.yaml
Comment (Zash) [reply helmfile/charts/prometheus-alerts/files/disk-perf.yaml]: > configurable from the overwrite configs

Should we?

File: helmfile/charts/prometheus-alerts/values.yaml
Comment (Zash) [reply helmfile/charts/prometheus-alerts/values.yaml:121]: @crssnd not `false` if it should be disabled for wc?

File: helmfile/charts/prometheus-alerts/templates/alerts/disk-perf.yaml
Comment (Zash) [inline helmfile/charts/prometheus-alerts/templates/alerts/disk-perf.yaml:2]: @crssnd this should allow disabling from config, right? (re comment on `files/disk-perf.yaml`)

File: helmfile/charts/prometheus-alerts/files/disk-perf.yaml
Comment (Zash) [reply helmfile/charts/prometheus-alerts/files/disk-perf.yaml]: Turned up to `for: 10m`.

File: helmfile/charts/prometheus-alerts/templates/alerts/disk-perf.yaml
Comment (crssnd) [reply helmfile/charts/prometheus-alerts/templates/alerts/disk-perf.yaml:2]: to be able to disable them from the overwrite configs you will need to add that option in this [file](https://github.com/elastisys/compliantkubernetes-apps/blob/main/helmfile/values/prometheus-alerts-sc.yaml.gotmpl#L31) and also the value in [sc-config.yaml](https://github.com/elastisys/compliantkubernetes-apps/blob/main/config/config/sc-config.yaml)

**Review by lucianvlad — COMMENTED** [review]
Most of the pipeline tests are failing !
Can those be fixed before merging?

File: helmfile/charts/prometheus-alerts/values.yaml
Comment (Zash) [reply helmfile/charts/prometheus-alerts/values.yaml:121]: Would this be more fitting under `.diskAlerts` ?

File: helmfile/charts/prometheus-alerts/templates/alerts/disk-perf.yaml
Comment (lucianvlad) [reply helmfile/charts/prometheus-alerts/templates/alerts/disk-perf.yaml:2]: LGTM unless @crssnd has anything to add

**Review by lucianvlad — APPROVED** [review]

**Comment by Zash** [comment]
FYI I moved the setting to `.prometheus.diskAlerts.perf` which seemed appropriate in my head and simplifies templating since `diskAlerts` is passed trough as is

**Review by crssnd — APPROVED** [review]

**Pull request merged**
